### ISSUE TITLE
[webxr][hit-test] Tests seem to be missing some intersections

### ISSUE THREAD

**Issue by Manishearth** [web-platform-tests/wpt#22898]
https://github.com/web-platform-tests/wpt/blob/f2e918abb4b8711afb28c8b98d98be3bffc9fa9a/webxr/hit-test/ar_hittest_subscription_refSpaces.https.html#L110-L130

In this test, the viewer is at (0, 1, 0) and looking straight ahead (the default XRRay). This should intersect with both of [the faces](https://github.com/web-platform-tests/wpt/blob/f2e918abb4b8711afb28c8b98d98be3bffc9fa9a/webxr/resources/webxr_test_constants_fake_world.js#L32-L33) in front of the viewer since `height` is 2.0 and `y=1` is where their midpoint is.

cc @bialpio

**Comment by bialpio** [comment]
Intersection with both of the faces results in exactly the same pose, so in Chrome's mock implementation I am deduplicating those kinds of results. Both faces that intersect the ray belong to the same mesh that is supposed to represent a single, contiguous surface, so it made no sense to me to return 2 results in this case. Let me know if you think we should revisit this approach.

**Comment by Manishearth** [comment]
This isn't mentioned in the spec anywhere. I think it's not a good idea to make the test dependent on such an implementation detail, perhaps the ray should be shifted up a little bit.

**Comment by Manishearth** [comment]
@bialpio do you agree with shifting the ray up a little bit? if not this needs to be a spec bug

**Comment by Manishearth** [comment]
We can also tweak the test to allow both one and two hit test results, and then tweak the spec to allow this optimization.

**Comment by bialpio** [comment]
Shifting it up SGTM, as long as it makes the manual calculations still feasible - I usually calculate the expected results by hand before codifying them in order to avoid setting the expected data to whatever makes the tests pass.

If we were to spec it out, some questions we'd have to answer:
- Is the intersection of ray with triangle defined on triangle's edges / vertices? if so, what should be the surface normal there? The intent was to represent a contiguous surface but approximated with triangles.
- What about faces with triangles that do not lie on the same plane? Fake world currently does that, but it doesn't have to be this way for all tests...
- Do regions form contiguous surface with their neighboring regions?

I think it's probably OK to dodge those for now and move the ray or the space relative to which it's expressed.

**Comment by Manishearth** [comment]
Yeah, most of the vectors are along the axes so shifting up is easy.

If we were to spec it out I'd imagine that we just spec that we filter out duplicates.

I'll fix this once https://github.com/web-platform-tests/wpt/pull/23350 lands to avoid test conflicts.
